Sources/UserDefaults/UserDefaults.swift

```diff
@@ -26,7 +26,7 @@ import Foundation
 public struct UserDefault<Value: Codable> {
   private let key: String
   private let defaultValue: Value
-  private var container: UserDefaults = .standard
+  private let container: UserDefaults = .standard
```

Comment:
If you made it let, this means that will not be able to edit the container if needed. The container is not constant but rather  a value that can be changed, it is include in the init but as an optional. So needs to be var

Reply:
right, that was my point. i just wanted to ask if it's possible to have an immutable container property inside your structure.

Reply:
Actully, you can if you wrote it like this `private let container: UserDefaults` without giving it a value. And initialize the value from the `init`. I reponed the PR for you edit it if you want @severeduck

Reply:
no, YOU edit it if you want :)

Reply:
because i dont.

Reply:
I thought you want to contribute

Reply:
>And initialize the value from the init
wouldn't that be great? you then pass a closure to the `init`. For your unit tests - you'll only need to pass another (mock/dummy) closure to see how the system reacts.

Reply:
i did contribute by giving some ideas. I'm not gonna write the code for you.